pkg/stub/handler.go

```diff
@@ -372,10 +508,27 @@ func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
 		if srcfg.Name != v1alpha1.SamplesResourceName {
 			return nil
 		}
+
+		// pattern is 1) come in with delete timestamp, event delete flag false
+		// 2) then after we remove finalizer, comes in with delete timestamp
+		// and event delete flag true
 		if event.Deleted {
-			logrus.Println("SampleResource deleted")
-			h.samplesResource = nil
-			return h.CreateDefaultResourceIfNeeded()
+			// possibly tell poller to stop
+			logrus.Info("A previous delete attempt has been successfully completed")
+			return nil
+		}
+		if srcfg.DeletionTimestamp != nil {
+			if h.NeedsFinalizing(srcfg) {
+				logrus.Println("Initiating finalizer processing for a SampleResource delete attempt")
+				h.RemoveFinalizer(srcfg)
+				h.CleanUpOpenshiftNamespaceOnDelete(srcfg)
```

Comment:
you also need to delete the configmap, right?  Either via this finalizer or via an ownerreference relationship to the samplesresource.